crates/nu-cli/src/commands/to_xml.rs

```diff
@@ -0,0 +1,139 @@
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::{ShellError};
+use nu_protocol::{Primitive, ReturnSuccess, Signature, SyntaxShape, UntaggedValue, Value};
+use std::io::Write;
+
+pub struct ToXML;
+
+#[derive(Deserialize)]
+pub struct ToXMLArgs {
+    pretty: Option<Value>,
+}
+
+#[async_trait]
+impl WholeStreamCommand for ToXML {
+    fn name(&self) -> &str {
+        "to xml"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("to xml").named(
+            "pretty",
+            SyntaxShape::Int,
+            "Formats the XML text with the provided indentation setting",
+            Some('p'),
+        )
+    }
+
+    fn usage(&self) -> &str {
+        "Convert table into .xml text"
+    }
+
+    async fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        to_xml(args, registry).await
+    }
+}
+
+pub fn write_xml_events<W: Write>(value: &Value, writer: &mut xml::writer::EventWriter<W>) -> Result<(), ShellError> {
+    Ok(match &value.value {
+        UntaggedValue::Primitive(Primitive::Boolean(b)) => {
+            writer.write(b.to_string().as_str()).unwrap();
```

Comment:
`.write()` returns a Result<()>. What's the point of Result type with no error?

Reply:
That is Type Alias
`type Result<T> = Result<T, EmitterError>;`
https://docs.rs/xml-rs/0.7.0/xml/writer/type.Result.html

So you should handle the error, either by using `expect` if you KNOW it won't fail or using something like `.map_err(|e| ShellError::labelled_error(format!("Something happened {:?}", e) ...`